File: 03.programmers/lv01/02.Python/05_2016_year.py
def solution(a, b):
    day = ["FRI","SAT","SUN","MON","TUE","WED","THU"]
    date_of_month = [31,29,31,30,31,30,31,31,30,31,30,31]
    return day[(sum([ date_of_month[i] for i in range(a-1)]) + b -1)%7]

input_a, input_b = 5, 24
print(solution(input_a,input_b))

File: ~2023/03.programmers/lv01/01.C++/05_2016_year.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

string solution(int a, int b) {
    string answer = "";
    vector<string> day = {"FRI","SAT","SUN","MON","TUE","WED","THU"};
    vector<int> date_of_month = {31,29,31,30,31,30,31,31,30,31,30,31};
    int d = -1;
    for(int i=0; i<a-1; i++)
        d +=date_of_month[i];
    d+=b;
    answer = day[d%7];
    return answer;
}
int main(){
    int a = 5, b = 24;
    cout<<solution(a,b)<<endl;
    return 0;
}